packages/pn-validator/src/ruleValidators/__test__/ArrayRuleValidator.test.ts

```diff
@@ -1,21 +1,29 @@
 import { vi } from 'vitest';
+
 import { ForEachElement } from '../../rules/ForEachElement';
 import { IsEmpty } from '../../rules/IsEmpty';
+import { Required } from '../../rules/Required';
 import { ArrayRuleValidator } from '../ArrayRuleValidator';
 
 const pushRuleMk = vi.fn();
 const dummyRuleValidator = new ArrayRuleValidator(pushRuleMk);
 
 vi.mock('../../rules/IsEmpty', () => {
-    return {
-        IsEmpty: vi.fn()
-    }
+  return {
+    IsEmpty: vi.fn(),
+  };
 });
 
 vi.mock('../../rules/ForEachElement', () => {
-    return {
-        ForEachElement: vi.fn()
-    }
+  return {
+    ForEachElement: vi.fn(),
+  };
+});
+
+vi.mock('../../rules/Required', () => {
```

Comment:
Remove the test about the required rule and leave only that is in the common rule validator. Do it for all tests.

After that add the check into the ValidatorBuilder.test.tsx